alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -42,7 +42,37 @@ void RunnableObject::stop()
         return;
     }
     ++_signalState;
-    _engine->editScheduler().schedule(std::bind(&RunnableObject::doTerminate, this));
+    _engine->editScheduler().schedule([this]() { doTerminate(); });
+}
+
+void RunnableObject::stop(RunningPlan* rp)
+{
+    if (!isActive(_signalState.load())) {
+        return;
+    }
+    ++_signalState;
+    if (rp->getParent()) {
+        if (!_inheritBlackboard) {
+            assert(rp->getParent()->getBasicPlan());
+            const auto& wrappers = rp->getParent()->getActiveState()->getConfAbstractPlanWrappers();
+            auto it = std::find_if(
+                    wrappers.begin(), wrappers.end(), [this](const auto& wrapper_ptr) { return wrapper_ptr->getAbstractPlan()->getName() == _name; });
+            assert(it != wrappers.end());
+
+            int64_t wrapperId = (*it)->getId();
+
+            BasicPlan* parentPlan = rp->getParent()->getBasicPlan();
+            const auto keyMapping = parentPlan->getKeyMapping(wrapperId);
+            auto terminateCall = [this, keyMapping, parentPlan = parentPlan]() {
+                assert(_blackboard);
+                keyMapping.setOutput(parentPlan->getBlackboard().get(), _blackboard.get());
```

Comment:
Can we set the output mapping the same time we set the input mapping (when initializing the behavior/plan)?

Reply:
The output must be copied on termination, so I don't see a another point where this could be done.

Reply:
I suppose the scheduling can only be done here, but the keyMapping lookup could have been cached from earlier to avoid going through all the wrappers etc again.  I suppose it would have to be cached from the scheduled init() call though, so maybe easier to just go through the wrappers again here.

Mostly just complaining about the duplicated code, so maybe the solution is to refactor out the code to find the appropriate keyMapping?